annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +338,983 @@ def test_validate_user_actions_invalid_states(
+    ("additional_filters"),
+    (({"file_name": ["file1"]}), ({"job_name": ["job1"]})),
+)
+def test_filter_tasks_db_no_files_or_jobs(
+    mock_session: Mock,
+    additional_filters: Dict[str, List[str]],
+):
+    files_by_name = {}
+    jobs_by_name = {}
+    expected_result = ([], {}, {})
+    request = TaskFilter()
+    tenant = "test_tenant"
+    token = "test_token"
+
+    with patch(
+        "annotation.tasks.services.map_request_to_filter",
+        return_value={"filters": [], "sorting": []},
+    ), patch(
+        "annotation.tasks.services.remove_additional_filters",
+        return_value=additional_filters,
+    ), patch(
+        "annotation.tasks.services.get_files_by_request",
+        return_value=files_by_name,
+    ), patch(
+        "annotation.tasks.services.get_jobs_by_name", return_value=jobs_by_name
+    ), patch(
+        "annotation.tasks.services.form_query",
+        return_value=(MagicMock(), MagicMock()),
+    ), patch(
+        "annotation.tasks.services.paginate",
+        return_value=([MagicMock()], MagicMock()),
+    ):
+
+        mock_query = MagicMock()
+        mock_session.query.return_value = mock_query
+        mock_query.filter.return_value = mock_query
+
+        result = filter_tasks_db(mock_session, request, tenant, token)
+
+        assert len(result[0]) == 2
+        assert result[1] == expected_result[1]
+        assert result[2] == expected_result[2]
+
+
+def test_create_tasks(mock_session: Mock):
+    tasks = [
+        {"user_id": 1, "file_id": 123, "pages": {1, 2}},
+        {"user_id": 2, "file_id": 456, "pages": {3}},
+    ]
+    job_id = 1
+    expected_user_ids = {1, 2}
+    with patch(
+        "annotation.tasks.services.update_files"
+    ) as mock_update_files, patch(
+        "annotation.tasks.services.update_user_overall_load"
+    ) as mock_update_user_overall_load:
+        mock_bulk_insert_mappings = MagicMock()
+        mock_session.bulk_insert_mappings = mock_bulk_insert_mappings
+        mock_session.flush = MagicMock()
+
+        create_tasks(mock_session, tasks, job_id)
+
+        assert mock_bulk_insert_mappings.call_count == 1
+        assert mock_update_files.call_count == 1
+
+        mock_update_user_overall_load.assert_has_calls(
+            [call(mock_session, user_id) for user_id in expected_user_ids],
+            any_order=True,
+        )
+
+
+def test_create_tasks_with_empty_tasks(mock_session: Mock):
+    tasks = []
+    job_id = 1
+    with patch(
+        "annotation.tasks.services.update_files"
+    ) as mock_update_files, patch(
+        "annotation.tasks.services.update_user_overall_load"
+    ) as mock_update_user_overall_load:
+        mock_bulk_insert_mappings = MagicMock()
+        mock_session.bulk_insert_mappings = mock_bulk_insert_mappings
+        mock_session.flush = MagicMock()
+
+        create_tasks(mock_session, tasks, job_id)
+
+        assert mock_bulk_insert_mappings.call_count == 1
+        assert mock_update_files.call_count == 1
```

Comment:
Again, should we check whether it was called with certain parameters?

Reply:
```suggestion
        mock_update_files.assert_called_once()
```
There is no apparent reason not to use assert_called_once() over assert call_count == 1